### PR TITLE
Add redirects for SYSLIB warnings

### PR DESCRIPTION
**Pull request by gewarren**
Some SYSLIB warnings are documented as a group in a single file, but the source code for doc help assumes a single article named with the error code. These redirects solve the problem.

Contributes to #35297 
Contributes to dotnet/runtime#85181

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/syslib-diagnostics/source-generator-overview.md](https://github.com/dotnet/docs/blob/11f82a820168152fe3506ec87fe172e9d087b564/docs/fundamentals/syslib-diagnostics/source-generator-overview.md) | [Source-generator diagnostics in .NET 6+](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/source-generator-overview?branch=pr-en-us-35301) |
| [docs/fundamentals/syslib-diagnostics/syslib1050-1069.md](https://github.com/dotnet/docs/blob/11f82a820168152fe3506ec87fe172e9d087b564/docs/fundamentals/syslib-diagnostics/syslib1050-1069.md) | [SYSLIB diagnostics for p/invoke source generation](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1050-1069?branch=pr-en-us-35301) |
| [docs/fundamentals/syslib-diagnostics/syslib1070-1089.md](https://github.com/dotnet/docs/blob/11f82a820168152fe3506ec87fe172e9d087b564/docs/fundamentals/syslib-diagnostics/syslib1070-1089.md) | [docs/fundamentals/syslib-diagnostics/syslib1070-1089](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1070-1089?branch=pr-en-us-35301) |
| [docs/fundamentals/syslib-diagnostics/syslib1090-1099.md](https://github.com/dotnet/docs/blob/11f82a820168152fe3506ec87fe172e9d087b564/docs/fundamentals/syslib-diagnostics/syslib1090-1099.md) | [docs/fundamentals/syslib-diagnostics/syslib1090-1099](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1090-1099?branch=pr-en-us-35301) |
| [docs/fundamentals/syslib-diagnostics/syslib1100-1118.md](https://github.com/dotnet/docs/blob/11f82a820168152fe3506ec87fe172e9d087b564/docs/fundamentals/syslib-diagnostics/syslib1100-1118.md) | [docs/fundamentals/syslib-diagnostics/syslib1100-1118](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1100-1118?branch=pr-en-us-35301) |

### CODE REVIEW

File: .openpublishing.redirection.fundamentals.json [hunk 1]
```diff
@@ -120,6 +120,206 @@
             "source_path_from_root": "/docs/fundamentals/syslib-diagnostics/syslib1004.md",
             "redirect_url": "/dotnet/fundamentals/syslib-diagnostics/syslib1001"
         },
+        {
```
Comment (danmoseley) [inline .openpublishing.redirection.fundamentals.json:123]: I think you should include 1030 upward. that's where the generator ID's start. if we're missing a 1030-1039 link we should create one.
see
https://github.com/dotnet/runtime/blob/main/docs/project/list-of-diagnostics.md?plain=1#L210

Comment (danmoseley) [reply .openpublishing.redirection.fundamentals.json:123]: also you could anticipate 1090-1118 which are missing but will appear. after all many of these in this PR don't exist yet eg 1046. I suggest just set up redirects for everything from 1030 to 1118

Comment (gewarren) [reply .openpublishing.redirection.fundamentals.json:151]: > I think you should include 1030 upward. that's where the generator ID's start.

Those IDs have their own pages so no redirects are needed - e.g. https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib1030.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

I like the consolidation to minimize the total number of pages. I think that will make it easier to maintain over time.

**Pull request merged**